frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import {
+    Box,
+    CircularProgress,
+    CircularProgressProps,
+    LinearProgress,
+    LinearProgressProps,
+    Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface ProgressBarProps {
+    linear?: boolean; //by default - false
+    showProgress?: boolean; //by default - false
+    value?: number; //progress value
+    defaultValue?: number; //default progress value
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear, showProgress, value, defaultValue } = props;
+
+    const [linearProgress, setLinearProgress] = useState(false);
+    const [progressVisible, setProgressVisible] = useState(false);
+
+    const val = useDynamicProperty(props.value, props.defaultValue, undefined);
+
+    useEffect(() => {
+        setLinearProgress((progress) => {
+            return props.linear !== undefined ? props.linear : progress;
+        });
+    }, [props.linear, linear]);
+
+    useEffect(() => {
+        setProgressVisible((progress_visible) => {
+            return props.showProgress !== undefined ? props.showProgress : progress_visible;
+        });
+    }, [props.showProgress, showProgress]);
+
+    //circular progress element
+    function CircularProgressWithLabel(props: CircularProgressProps & { value: number }) {
```

Comment:
I don't see the point of defining these *WithLabel functions as they are used only once.
I'd rather see all the JSX in one place (the return)

Reply:
These functions were in the MUi docs and were given as separate functions because we can't render both the linear and circular elements at once.

Reply:
yes they are examples
We don't want to render everything at once but there will be `if/else` blocks

Reply:
no function at all would be fine

Reply:
Actually, I was trying to do it without the functions, but actually, it's creating an issue that then we'll not be able to use `CircularProgressProps` or `LinearProgressProps` without a function? I'm trying to fix this.

Reply:
@FredLL-Avaiga, could you please guide that how can this be done without using these 2 functions? I'm trying, but I'm not able to solve this completely.

Reply:
this is how I would do it 

```
import React from "react";
import Box from "@mui/material/Box";
import CircularProgress from "@mui/material/CircularProgress";
import LinearProgress from "@mui/material/LinearProgress";
import Typography from "@mui/material/Typography";

import { useDynamicProperty } from "../../utils/hooks";

interface ProgressBarProps {
    linear?: boolean; //by default - false
    showValue?: boolean; //by default - false
    value?: number; //progress value
    defaultValue?: number; //default progress value
}

const Progress = (props: ProgressBarProps) => {
    const { linear, showValue } = props;

    const value = useDynamicProperty(props.value, props.defaultValue, undefined);

    return showValue && value !== undefined ? (
        linear ? (
            <Box sx={{ display: "flex", alignItems: "center" }}>
                <Box sx={{ width: "100%", mr: 1 }}>
                    <LinearProgress variant="determinate" value={value} />
                </Box>
                <Box sx={{ minWidth: 35 }}>
                    <Typography variant="body2" color="text.secondary">{`${Math.round(value)}%`}</Typography>
                </Box>
            </Box>
        ) : (
            <Box sx={{ position: "relative", display: "inline-flex" }}>
                <CircularProgress variant="determinate" value={value} />
                <Box
                    sx={{
                        top: 0,
                        left: 0,
                        bottom: 0,
                        right: 0,
                        position: "absolute",
                        display: "flex",
                        alignItems: "center",
                        justifyContent: "center",
                    }}
                >
                    <Typography variant="caption" component="div" color="text.secondary">
                        {`${Math.round(value)}%`}
                    </Typography>
                </Box>
            </Box>
        )
    ) : linear ? (
        <LinearProgress variant={value === undefined ? "indeterminate": "determinate"} value={value} />
    ) : (
        <CircularProgress variant={value === undefined ? "indeterminate": "determinate"} value={value} />
    );
};

export default Progress;

```